### PR TITLE
Add save-graph path option to write modified graph to

### PR DESCRIPTION
**Pull request by catch-point**

### CODE REVIEW

File: src/noflo-nodejs.coffee [hunk 1]
```diff
@@ -143,6 +146,10 @@ startServer = (program, defaultGraph) ->
     tracer.dumpFile null, (err, fname) ->
       console.log 'Wrote flowtrace to:', fname
 
+  if program['save-graph']
+    defaultGraph.on 'endTransaction', (id, metadata) ->
+      fs.writeFileSync program['save-graph'], JSON.stringify(defaultGraph, null, 2)
```
Comment (dbooth-boston) [inline src/noflo-nodejs.coffee:151]: If this could be pretty-printed JSON it would be nice.

**Review by dbooth-boston — APPROVED** [review]
LGTM subject to one minor suggestion.  No need for re-review.

File: src/noflo-nodejs.coffee (hunk 1 above)
Comment (dbooth-boston) [reply src/noflo-nodejs.coffee:151]: My comment is obsolete.  Apparently the JSON is already formatted enough.

**Comment by glenna-m** [comment]
Two requests: 
1. Could we add support to create the graph file if it does not already exist? 
2. The current documentation assumes the user is just using noflo-nodejs, not doing development.  Can we add a section that documents the steps for setting up an environment for doing development and running with that?

**Comment by glenna-m** [comment]
LGTM

**Pull request merged**